ostd/src/sync/rwmutex.rs

```diff
@@ -377,7 +377,7 @@ impl<T: ?Sized, R: Deref<Target = RwMutex<T>>> Deref for RwMutexUpgradeableGuard
 impl<T: ?Sized, R: Deref<Target = RwMutex<T>>> Drop for RwMutexUpgradeableGuard_<T, R> {
     fn drop(&mut self) {
         let res = self.inner.lock.fetch_sub(UPGRADEABLE_READER, Release);
-        if res == 0 {
+        if res == UPGRADEABLE_READER {
```

Comment:
I don't think that this modification is correct. Yes, it wakes up any concurrent threads are waiting to acquire an upgradable read lock (`res == UPGRADEABLE_READER`). But it fails to wake up any concurrent threads are waiting to acquire a write lock (`res == 0`). So I think the condition should be `res == 0 || res == UPGRADEABLE_READER`.

Reply:
The behavior of `fetch_sub()` might cause confusion. It returns the previous value before subtraction. So if `res == 0`, then after subtracting, the `lock` value would be `-UPGRADEABLE_READER`, which is invalid. When `res == UPGRADEABLE_READER`, it indicates the last reader is releasing, so waking up waiting threads.

Since waiting threads are competing for an exclusive resource (write lock or upgradable read lock), waking up only one thread can prevent the "thundering herd" problem.

Reply:
> It returns the previous value before subtraction. So if res == 0, then after subtracting, the lock value would be -UPGRADEABLE_READER, which is invalid. 

Yes, you are right. Thanks for pointing it out! Then I have no problem with this PR.